reactivesocket-core/src/main/java/io/reactivesocket/ReactiveSocket.java

```diff
@@ -46,6 +46,23 @@
     double availability();
 
     /**
+     * Close this {@code ReactiveSocket} upon subscribing to the returned {@code Publisher}
+     *
+     * <em>This method is idempotent and hence can be called as many times at any point with same outcome.</em>
+     *
+     * @return A {@code Publisher} that completes when this {@code ReactiveSocket} close is complete.
+     */
+    Publisher<Void> close();
+
+    /**
+     * Returns a {@code Publisher} that completes when this {@code ReactiveSocket} is closed. A {@code ReactiveSocket}
+     * can be closed by explicitly calling {@link #close()} or when the underlying transport connection is closed.
+     *
+     * @return A {@code Publisher} that completes when this {@code ReactiveSocket} close is complete.
+     */
+    Publisher<Void> closeNotifier();
```

Comment:
Not strongly opinionated about that, but what do you think about the name `onClose`?